src/Framework/BuildEngineInterface.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// Exposes build engine functionality that was made available in newer versions of MSBuild.
+    /// </summary>
+    /// <remarks>
+    /// Make all members virtual but not abstract, ensuring that implementations can override them and external implementations
+    /// won't break when the class is extended with new members. This base implementation should be throwing <see cref="NotImplementedException"/>.
+    /// </remarks>
+    [Serializable]
+    public abstract class BuildEngineInterface
```

Comment:
note to self when reviewing: Mentioned offline: naming here. Alternate proposal in a linked bug?

Reply:
Yep, `EngineServices`. Or maybe `BuildEngineServices`.

Reply:
`BuildEngineBase`?

Reply:
Intersecting my and Rainer's preferences, it looks like the winning option is `EngineServices` and we would also be fine with `BuildEngineInterface` (or just `EngineInterface`). @Forgind, please let me know if you feel strongly about your proposal or the runner-up. If not, I'll proceed with renaming to `EngineServices` and mark the PR for merge. Thank you!